x-pack/plugins/cloud_security_posture/server/routes/compliance_dashboard/get_trends.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient, SearchResponse } from 'kibana/server';
+import { BENCHMARK_SCORE_INDEX_PATTERN } from '../../../common/constants';
+import { Stats } from '../../../common/types';
+import { calculatePostureScore } from './get_stats';
+
+export interface ScoreTrendDoc {
+  '@timestamp': string;
+  total_findings: number;
+  passed_findings: number;
+  failed_findings: number;
+  score_by_cluster_id: Record<
+    string,
+    {
+      total_findings: number;
+      passed_findings: number;
+      failed_findings: number;
+    }
+  >;
+}
+
+export const getTrendsAggsQuery = () => ({
+  index: BENCHMARK_SCORE_INDEX_PATTERN,
+  size: 5,
+  sort: {
+    '@timestamp': {
+      order: 'desc',
+    },
+  },
+});
+
+export type Trends = Array<{
+  timestamp: string;
+  summary: Stats;
+  clusters: Record<string, Stats>;
+}>;
+
+type CorrectSearchResponse<TDocument> = Record<'body', SearchResponse<TDocument>>;
+
+export const getTrendsFromQueryResult = (scoreTrendDocs: ScoreTrendDoc[]): Trends =>
+  scoreTrendDocs
+    .map((data) => ({
+      timestamp: data['@timestamp'],
+      summary: {
+        totalFindings: data.total_findings,
+        totalFailed: data.failed_findings,
+        totalPassed: data.passed_findings,
+        postureScore: calculatePostureScore(data.passed_findings, data.failed_findings),
+      },
+      clusters: Object.fromEntries(
+        Object.entries(data.score_by_cluster_id).map(([clusterId, cluster]) => [
+          clusterId,
+          {
+            totalFindings: cluster.total_findings,
+            totalFailed: cluster.failed_findings,
+            totalPassed: cluster.passed_findings,
+            postureScore: calculatePostureScore(cluster.passed_findings, cluster.failed_findings),
+          },
+        ])
+      ),
+    }))
+    .sort((a, b) => new Date(a.timestamp).getDate() - new Date(b.timestamp).getDate());
```

Comment:
This sorting is a bit doubled, the query should already return a sorted array. but this allows me to test `getTrendsFromQueryResult` better to make sure this function always returns a sorted trend. 
Since this is happening synchronically I think it's worth

Reply:
I tend to disagree, this function's purpose is not to sort by date, that is something elasticsearch does for you, and there is no point in double sorting, it provides no value. Sorting should not be tested in the scope of this method, the way to test sorting is via an integration test that uses elasticsearch